pgx-pg-sys/src/lib.rs

```diff
@@ -167,6 +167,9 @@ pub trait PgNode {
         // this is only implemented for things known to be "Nodes"
         unsafe { display_node_impl(NonNull::from(self).cast()) }
     }
+
+    /// Traverse the object graph, calling the callback at any `Node`s found as children of this object.
+    fn traverse<T>(&mut self, walker_fn: fn(*mut Node, &mut T) -> (), context: &mut T);
```

Comment:
I'm not terribly excited by there not being a default implementation that's valid, and by the `walker_fn` using `*mut Node`. I would prefer that we rely less on macro-generated code for implementations, as it's harder to audit, not more.